src/claimtrie.cpp

```diff
@@ -1634,156 +1095,107 @@ bool CClaimTrieCache::addClaimToQueues(const std::string& name, CClaimValue& cla
     return true;
 }
 
-bool CClaimTrieCache::removeClaimFromQueue(const std::string& name, const COutPoint& outPoint, CClaimValue& claim) const
+bool CClaimTrieCache::removeClaimFromQueue(const std::string& name, const COutPoint& outPoint, CClaimValue& claim)
 {
-    queueNameType::iterator itQueueNameRow = getQueueCacheNameRow(name, false);
-    if (itQueueNameRow == claimQueueNameCache.end())
-    {
-        return false;
-    }
+    queueNameType::iterator itQueueNameRow = getQueueCacheRow(name, claimQueueNameCache, false);
+    if (itQueueNameRow == claimQueueNameCache.end()) return false;
     queueNameRowType::iterator itQueueName;
     for (itQueueName = itQueueNameRow->second.begin(); itQueueName != itQueueNameRow->second.end(); ++itQueueName)
     {
-        if (itQueueName->outPoint == outPoint)
-        {
-            break;
-        }
-    }
-    if (itQueueName == itQueueNameRow->second.end())
-    {
-        return false;
+        if (itQueueName->outPoint == outPoint) break;
     }
-    claimQueueType::iterator itQueueRow = getQueueCacheRow(itQueueName->nHeight, false);
-    if (itQueueRow != claimQueueCache.end())
-    {
+    if (itQueueName == itQueueNameRow->second.end()) return false;
+    claimQueueType::iterator itQueueRow = getQueueCacheRow(itQueueName->nHeight, claimQueueCache, false);
+    if (itQueueRow != claimQueueCache.end()) {
         claimQueueRowType::iterator itQueue;
-        for (itQueue = itQueueRow->second.begin(); itQueue != itQueueRow->second.end(); ++itQueue)
-        {
-            if (name == itQueue->first && itQueue->second.outPoint == outPoint)
-            {
-                break;
+        for (itQueue = itQueueRow->second.begin(); itQueue != itQueueRow->second.end(); ++itQueue) {
+            if (name == itQueue->first && itQueue->second.outPoint == outPoint) {
+                std::swap(claim, itQueue->second);
+                itQueueNameRow->second.erase(itQueueName);
+                itQueueRow->second.erase(itQueue);
```

Comment:
What does this line do if getQueueCacheRow actually called Read?

Reply:
CDBWrapper::Read does not automatically add to the cache as far as I can tell.

Reply:
getQueueCacheRow put it to cache, line:1079, this is safe, it ease from cache.